src/sentry/tasks/relocation.py

```diff
@@ -131,9 +145,197 @@
+        request RPC call is received, and is immediately packaged into a `ControlOutbox`, so that we
+        may robustly forward it to the exporting region. This task successfully completing causes
+        the RPC to successfully return to the sender, allowing the calling `uploading_start` celery
+        task to finish successfully as well.
+    04. (CS) .../receiver/outbox/control.py::process_relocation_request_new_export: Whenever an
+        outbox draining attempt occurs, this code will be called to forward the proxied call into
+        the exporting region.
+    05. The `DBBackedExportService::request_new_export` call is sent over the wire from the control
+        silo to the exporting region.
+    06. (ER) .../relocation_export/impl.py::DBBackedRelocationExportService::request_new_export: The
+        request RPC call is received, and immediately schedules the `reply_with_remote_export`
+        celery task, which uses an exponential backoff algorithm to try and create an encrypted
+        tarball containing an export of the requested org slug. This data is written as a file to
+        this region's relocation-specific GCS bucket, and the response is immediately packaged into
+        a `RegionOutbox`, so that we may robustly attempt to send it at drain-time.
+    07. (ER) .../receiver/outbox/region.py::process_relocation_reply_with_export: Whenever an outbox
+        draining attempt occurs, this code will be called to read the saved export data from the
+        local GCS bucket, package it into an RPC call, and send it back to the proxy.
+    08. The `ProxyingRelocationExportService::reply_with_export` call is sent over the wire from the
+        exporting region back to the control silo.
+    09. (CS) .../relocation_export/impl.py::ProxyingRelocationExportService::reply_with_export: The
+        request RPC call is received, and is immediately packaged into a `ControlOutbox`, so that we
+        may robustly forward it back to the requesting region. To ensure robustness, the export data
+        is saved to a local file, so that outbox drain attempts can read it locally without needing
+        to make their own nested RPB calls.
+    10. (CS) .../receiver/outbox/control.py::process_relocation_reply_with_export: Whenever an
+        outbox draining attempt occurs, this code will be called to read the export data from the
+        local relocation-specific GCS bucket, then forward it into the requesting region.
+    11. The `DBBackedExportService::reply_with_export` call is sent over the wire from the control
+        silo back to the requesting region.
+    12. (RR) .../relocation_export/impl.py::DBBackedRelocationExportService::reply_with_export:
+        We've made it all the way back! The export data gets saved to a `RelocationFile` associated
+        with the `Relocation` that originally triggered `uploading_start`, and the next task in the
+        sequence (`uploading_complete`) is scheduled.
+    """
```

Comment:
Did you considering having the relocation process start in control? A sequence diagram like

```mermaid
sequenceDiagram

participant User
participant Destination
participant Control
participant Source

User ->> Control : move acme from us to de
Control ->> Control : save outbox for export request
Control ->> Destination : get public key for export
Control ->> Source : request export for slug=acme
Source ->> Source : export organization
Source ->> Control : send export blob
Control ->> Control : save blob and outbox
Control ->> Destination : send export blob and start import
Destination ->> Destination : save outbox + blob
Destination ->> Destination : process import
```

This division would have one fewer outbox driven step, but is otherwise mostly the same 🤷

Reply:
I think we still need an outbox after `export organization` and before `export blob`, since we don't want to have to redo the work of creating the export if the send call fails? I think it would end up with the same number of outboxes in that case.

More generally: I considered this as an option, but determined that it's a bit easier to start things on the "region where the relocation will go side", since that is where regular relocations go as well, and where the UUID-generating `Relocation` model lives. If we start it from `Control`, then we don't have the ID until we hit the requesting region for the first time with the `get public key for export` call.

Reply:
Also, just my own bias here, but the control flow feels a bit easier to reason about for me with the first diagram: we always move our request left to right (RR -> CS -> ER), generate the response, then move it back right to left (RR <- CS <- ER).

Reply:
> Also, just my own bias here, but the control flow feels a bit easier to reason about for me with the first diagram:

That is fair, the chevron does have a simplicity to it.